subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java

```diff
@@ -111,12 +111,19 @@ default void visitImplementations(ImplementationVisitor visitor) {
     }
 
     /**
-     * Visit regular inputs of the work.
+     * Visit identity inputs of the work.
+     *
+     * These are inputs that are passed to {@link #identify(Map, Map)} to calculate the identity of the work.
+     * These are more expensive to calculate than regular inputs as they need to be calculated even if the execution of the work is short circuited by an identity cache.
+     * They also cannot reuse snapshots taken during previous executions.
+     * Because of these reasons only capture inputs as identity if they are actually used to calculate the identity of the work.
      */
     default void visitIdentityInputs(InputVisitor visitor) {}
 
     /**
      * Visit regular inputs of the work.
+     *
+     * Regular inputs are inputs that are not used to calculate the identity of the work, but used to check up-to-dateness or to calculate the cache key.
```

Comment:
Can you clarify if one need to visit inputs which are part of the identity input as well here if they are regular inputs? Maybe this method should be called `visitInputs` only.

Reply:
I clarified.